app/routes/companyInterest/components/CompanyInterestPage.tsx

```diff
@@ -291,18 +142,36 @@ const EventBox = ({
   fields: any;
   language: string;
 }): ReactNode => (
-  <Flex column className={styles.checkboxWrapper}>
-    {fields.map((key, index) => (
-      <Field
-        key={`events[${index}]`}
-        name={`events[${index}].checked`}
-        label={EVENT_TYPES[eventToString(key)][language]}
-        type="checkbox"
-        component={CheckBox.Field}
-        normalize={(v) => !!v}
-      />
-    ))}
-  </Flex>
+  <FormSpy subscription={{ values: true }}>
+    {(props) => {
+      const newFields = fields.map((field) => field);
```

Comment:
Perhaps leave a comment explaining this. It looks a bit unnecessary if you're not too familiar with `react-final-form`.
I also think maybe "filteredFields" would be a more descriptive name for this variable, as nothing new is being added to the fields.